disnake/message.py

```diff
@@ -1538,6 +1539,25 @@ def system_content(self) -> Optional[str]:
         if self.type is MessageType.guild_incident_report_false_alarm:
             return f"{self.author.name} resolved an Activity Alert."
 
+        if self.type is MessageType.poll_result:
+            poll_result_embed = self.embeds[0]
+            get_field: Callable[[str], Optional[_EmbedFieldProxy]] = lambda field_name: utils.get(
```

Comment:
instead of iterating through the list of fields for every single `get_field()` call, it would be better to create a lookup dict from `embed._fields` upfront and then have `O(1)` access to fields